compiler/noirc_frontend/src/hir/def_map/module_data.rs

```diff
@@ -123,15 +123,16 @@ impl ModuleData {
     pub fn import(
         &mut self,
         name: Ident,
+        visibility: ItemVisibility,
         id: ModuleDefId,
         is_prelude: bool,
     ) -> Result<(), (Ident, Ident)> {
         // Empty spans could come from implicitly injected imports, and we don't want to track those
-        if name.span().start() < name.span().end() {
+        if visibility == ItemVisibility::Private && name.span().start() < name.span().end() {
```

Comment:
```suggestion
        if visibility != ItemVisibility::Public && name.span().start() < name.span().end() {
```

We should be able to find unused `pub(crate)` imports as well.

Reply:
I was thinking about this today. I think we could even detect unused functions (or unused anything) in a similar way, but I'll leave that for a separate PR (I'll push the pub(crate) use change, though!)